### PR TITLE
Rework the result for the session search to contain a single result per user sessions

### LINKED ISSUE #29203: Revisit SessionsResource#realmSessions as it current loads all sessions into memory
On a realm level, an admin can search the sessions with a string restricting it by user, client ID and IP address. 

Some of the logic should change as it should take the session ID and the offline/online flag to determine unique entries. In addition to that, for a lot of persistent sessions in the database, the searching should leverage more of the database capabilities and load fewer items to memory.

### PR DESCRIPTION
**Pull request by ahus1**
Closes #29203

This now returns only one entry per user session id and offline flag. A user session can then contain more than once client session, and both clients are presented in the UI. In the UI, the user can then log out the user session as before. As they are not logging out a single client session, this is IMHO how the UI should behave.

### CODE REVIEW

File: rest/admin-ui-ext/src/main/java/org/keycloak/admin/ui/rest/SessionsResource.java [hunk 1]
```diff
@@ -64,49 +63,125 @@ public Stream<SessionRepresentation> realmSessions(@QueryParam("type") @DefaultV
...
+            if (!search.equals("")) {
+                result = result.filter(s -> s.getUsername().contains(search) || s.getIpAddress().contains(search)
+                                            || s.getClients().values().stream().anyMatch(c -> c.contains(search)));
+            }
+            return result.skip(first).limit(max);
+        } else {
+            ClientModel client = null;
+            UserModel user = null;
+            String ipAddress;
+            if (!search.isEmpty()) {
```
Comment (hmlnarik) [inline rest/admin-ui-ext/src/main/java/org/keycloak/admin/ui/rest/SessionsResource.java:102]: Should this be rather `StringUtil.isNotBlank` and using `search.trim()` for substring searches? (Similarly to existing code above)

File: rest/admin-ui-ext/src/main/java/org/keycloak/admin/ui/rest/SessionsResource.java [hunk 2]
```diff
@@ -64,49 +63,125 @@ public Stream<SessionRepresentation> realmSessions(@QueryParam("type") @DefaultV
...
+                user = session.users().getUserByUsername(realm, search);
+                if (user != null) {
+                    if (!auth.users().canView(user)) {
+                        return Stream.empty();
```
Comment (hmlnarik) [inline rest/admin-ui-ext/src/main/java/org/keycloak/admin/ui/rest/SessionsResource.java:106]: Should this be returning an empty stream or just ignoring the user search? Suppose a search with `broker` string; existing user `broker` would be sufficient to return an empty stream while there could be numerous `broker` client sessions.

File: rest/admin-ui-ext/src/main/java/org/keycloak/admin/ui/rest/SessionsResource.java [hunk 3]
```diff
@@ -64,49 +63,125 @@ public Stream<SessionRepresentation> realmSessions(@QueryParam("type") @DefaultV
...
+                        case REGULAR:
+                            return session.sessions().getUserSessionsStream(realm, clientModel)
+                                    .map(s -> toRepresentation(s, REGULAR));
+                        case OFFLINE:
+                            return session.sessions()
+                                    .getOfflineUserSessionsStream(realm, clientModel, null, null)
+                                    .map(s -> toRepresentation(s, OFFLINE));
+                    }
+                    return Stream.<SessionRepresentation>builder().build();
+                }).distinct();
```
Comment (hmlnarik) [inline rest/admin-ui-ext/src/main/java/org/keycloak/admin/ui/rest/SessionsResource.java:173]: The `distinct()` could be applied on line 180 to optimize this op for filtered out entries.

**Review by hmlnarik — COMMENTED** [review]
Thanks @ahus1. Couple of comments left for further discussion.

File: rest/admin-ui-ext/src/main/java/org/keycloak/admin/ui/rest/SessionsResource.java [hunk 4]
```diff
@@ -64,49 +63,125 @@ public Stream<SessionRepresentation> realmSessions(@QueryParam("type") @DefaultV
...
+                    case REGULAR:
+                        return session.sessions().getUserSessionsStream(realm, clientModel)
+                                .map(s -> toRepresentation(s, REGULAR));
+                    case OFFLINE:
+                        return session.sessions()
+                                .getOfflineUserSessionsStream(realm, clientModel, null, null)
+                                .map(s -> toRepresentation(s, OFFLINE));
+                }
+                return Stream.<SessionRepresentation>builder().build();
+            }).distinct();
```
[the remark this answers is not in the document]
Comment (ahus1) [reply rest/admin-ui-ext/src/main/java/org/keycloak/admin/ui/rest/SessionsResource.java]: done

File: rest/admin-ui-ext/src/main/java/org/keycloak/admin/ui/rest/SessionsResource.java (hunk 1 above)
Comment (ahus1) [reply rest/admin-ui-ext/src/main/java/org/keycloak/admin/ui/rest/SessionsResource.java]: done

File: rest/admin-ui-ext/src/main/java/org/keycloak/admin/ui/rest/SessionsResource.java (hunk 2 above)
Comment (ahus1) [reply rest/admin-ui-ext/src/main/java/org/keycloak/admin/ui/rest/SessionsResource.java]: this has been reworked, I'm sticking with the original logic

File: rest/admin-ui-ext/src/main/java/org/keycloak/admin/ui/rest/SessionsResource.java (hunk 3 above)
Comment (ahus1) [reply rest/admin-ui-ext/src/main/java/org/keycloak/admin/ui/rest/SessionsResource.java]: this has been reworked, I'm sticking with the original logic

**Comment by ahus1** [comment]
@hmlnarik - this is now ready for review when you have the time. IMHO the unstable tests are unrelated to this change. Thanks!

**Comment by hmlnarik** [comment]
@ahus1 The change did not start Javascript CI tests since those are skipped. Could you please add 
```
rest/admin-ui-ext js
```
to `.github/actions/conditional/conditions`?

**Comment by ahus1** [comment]
@hmlnarik - added the change, and now the JS tests ran and are all green. Other parts of the build are still running, still I assume they will be green as they have been green before.

**Review by hmlnarik — APPROVED** [review]
Thanks @ahus1! Approved.

**Pull request merged**